charts/nd-common/templates/_istio.tpl

```diff
@@ -102,10 +102,10 @@ Usage: To setup the values follow the regular k8s resources structure under the
        memory: 60Mi
 */ -}}
 {{- with .Values.istio.resources }}
-sidecar.istio.io/proxyCPU: {{ .requests.cpu | default "100m" | quote }}
-sidecar.istio.io/proxyCPULimit: {{ .limits.cpu | default "2" | quote }}
-sidecar.istio.io/proxyMemory: {{ .requests.memory | default "128Mi" | quote }}
-sidecar.istio.io/proxyMemoryLimit: {{ .limits.memory | default "1Gi" | quote }}
+sidecar.istio.io/proxyCPU: {{ .requests.cpu | quote }}
```

Comment:
So the problem I saw is because of the values hierarchy, if you set istio.resources.requests - it comlains that istio.resources.limits does not exist when trying to evaluate limits.cpu - and that makes sense because there are no keys to search in null.
I could in theory add code to search for values gradually but it will not be clean - that's the cost of adding hierarchy to values.

Reply:
I get it ... I think you may just need a bit of a nested `with` statements... or create a default variable and then populate it. Something like:

(this is untested.. but rough idea)
```
{{- with .Values.istio.resources }}
{{- $requests := default (dict "empty" "dict") .requests }}
{{- $limits := default (dict "empty" "dict") .limits }}
sidecar.istio.io/proxyCPU: {{ $requests.cpu | default "2" | quote }}
sidecar.istio.io/proxyCPULimit: {{ $limits.cpu | default "2" | quote }}
sidecar.istio.io/proxyMemory: {{ $requests.memory | default "128Mi" | quote }}
sidecar.istio.io/proxyMemoryLimit: {{ $limits.memory | default "1Gi" | quote }}
{{- end }}
```

Reply:
That's a good idea, maybe a bit cleaner than what I did - let me change it accordingly.